### PR TITLE
[Distributed] Support multiple servers

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
We need to allow users to run multiple servers on a machine. This can improve the performance of kvstore and distributed sampling.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: tests/distributed/test_dist_graph_store.py [hunk 1]
```diff
@@ -203,15 +205,16 @@ def check_server_client(shared_mem):
...
+                                                 num_clients, shared_mem))
         serv_ps.append(p)
         p.start()
 
     cli_ps = []
-    for cli_id in range(1):
+    for cli_id in range(num_clients):
         print('start client', cli_id)
-        p = ctx.Process(target=run_client, args=(graph_name, cli_id, g.number_of_nodes(),
+        p = ctx.Process(target=run_client, args=(graph_name, 0, num_clients, g.number_of_nodes(),
```
Comment (classicsong) [inline tests/distributed/test_dist_graph_store.py:217]: why args=(graph_name, 0, num_clients, g.number_of_nodes(), 
not args=(graph_name, cli_id, g.number_of_nodes()

File: python/dgl/distributed/partition.py [hunk 2]
```diff
@@ -138,6 +138,9 @@ def load_partition(conf_file, part_id):
     assert EID in graph.edata, "the partition graph should contain edge mapping to global edge Id"
 
     gpb, graph_name = load_partition_book(conf_file, part_id, graph)
+    nids = F.boolean_mask(graph.ndata[NID], graph.ndata['inner_node'])
+    partids = gpb.nid2partid(nids)
+    assert np.all(F.asnumpy(partids == part_id)), 'load a wrong partition'
     return graph, node_feats, edge_feats, gpb, graph_name
```
Comment (classicsong) [inline python/dgl/distributed/partition.py:144]: What will happen if there are multiple features?

Comment (zheng-da) [reply python/dgl/distributed/partition.py:144]: node_feats and edge_feats are dictionaries, so they can handle multiple features.

File: tests/distributed/test_dist_graph_store.py (hunk 1 above)
Comment (zheng-da) [reply tests/distributed/test_dist_graph_store.py:217]: 0 is partition id. partition id and client id are different. one machine has only one partition, but one machine can run multiple servers and multiple clients.

File: python/dgl/distributed/rpc.py [hunk 3]
```diff
@@ -766,7 +766,9 @@ def send_requests_to_machine(target_and_requests):
         service_id = request.service_id
         msg_seq = incr_msg_seq()
         client_id = get_rank()
-        server_id = target
+
+        server_id = random.randint(target*get_num_server_per_machine(),
```
Comment (classicsong) [inline python/dgl/distributed/rpc.py:770]: Randomly select a backup server to sent the request?

Comment (zheng-da) [reply python/dgl/distributed/rpc.py:770]: yes. `send_request_to_machine` does the same thing.

**Pull request merged**